core/src/main/java/hudson/model/AbstractCIBase.java

```diff
@@ -89,17 +89,31 @@ protected void killComputer(Computer c) {
         c.kill();
     }
 
-    /* =================================================================================================================
-    * Package-protected, but accessed API
-    * ============================================================================================================== */
+    private final Set<String> disabledAdministrativeMonitors = Collections.synchronizedSet(new HashSet<>());
 
-    /*package*/ final CopyOnWriteArraySet<String> disabledAdministrativeMonitors = new CopyOnWriteArraySet<>();
+    /**
+     * Get the disabled administrative monitors
+     *
+     * @since TODO
+     */
+    public Set<String> getDisabledAdministrativeMonitors(){
+        synchronized (disabledAdministrativeMonitors) {
```

Comment:
This can just return the object. The synchronization will happen either implicitly through calls to the `Set` methods or when we synchronize on the object explicitly in the other two methods.